src/sqlfluff/core/parser/segments/base.py

```diff
@@ -343,7 +343,7 @@ def raw_segments(self) -> List["BaseSegment"]:
         return self.get_raw_segments()
 
     @cached_property
-    def raw_segments_upper(self) -> Optional[str]:
+    def first_non_whitespace_raw_segments_upper(self) -> Optional[str]:
```

Comment:
The name of this property was very deceptive. Updated it to reflect what it actually does.